### ISSUE TITLE
Make `$next` argument required in `MiddlewareInterface`

### ISSUE THREAD

**Issue by weierophinney** [zendframework/zend-stratigility#15]
With the current `MiddlewareInterface` definition, developers must first validate that `$next` is non-empty before calling it:

``` php
return ($next ? $next($req, $res) : $res);
```

This is confusing, and error-prone.

For version 2, the argument should be required.

**Comment by sandrokeil** [comment]
I'm just playing around with Middleware and if `$next` is required, what should I use for the `$next` parameter in the return definition? An empty callable?

```php
return $next($req, $res, /* What value makes sense? */);
```

**Comment by snapshotpl** [comment]
@sandrokeil nothing. Incomming `$next` it's not real next middleware. It's just callable which inside call next middleware.

**Comment by sandrokeil** [comment]
@snapshotpl Alright. Thank you very much. So it's a convention that, `$next` is not an instance of `MiddlewareInterface`. Otherwise it could leads to errors if `$next` is mandatory.

**Comment by weierophinney** [comment]
@sandrokeil Within Stratigility, it's a *queue* of middleware, and calling it iterates to the next and invokes it, *passing itself as an argument!* (Or, in the case that the queue is exhausted, passing the final handler.)

By *not* typehinting, we allow the signature to work with other paradigms, where it may actually be middleware.

**Comment by gabrielsch** [comment]
what about all ides/editors throwing "unused variable" warning? 😞

**Comment by xtreamwayz** [comment]
This could be closed since it is already implemented in 2.0-dev.

**Comment by weierophinney** [comment]
Closed; this is implemented on current develop branch, targeting 2.0.0 (which should drop this week).
